inst/report/scripts/generate_report.R

```diff
@@ -0,0 +1,705 @@
+
+    # Update settings based on upload type
+    updateUploadType <- function(upload_type) {
+        switch(upload_type,
+               "Fasta" = {
+                   resetSettings()
+                   acc_homology_analysis <<- TRUE
+                   acc_da_analysis <<- TRUE
+               },
+               "AccNum" = {
+                   resetSettings()
+                   acc_homology_analysis <<- TRUE
+                   acc_da_analysis <<- TRUE
+               },
+               "MSA" = {
+                   resetSettings()
+                   acc_homology_analysis <<- TRUE
+                   acc_da_analysis <<- TRUE
+               },
+               "BLAST Output" = {
+                   resetSettings()
+                   acc_phylogeny_analysis <<- TRUE
+               },
+               "InterProScan Output" = {
+                   resetSettings()
+                   acc_da_analysis <<- TRUE
+               }
+        )
+    }
+
+    updateUploadType(upload_type)
+
+    ####### File Upload Functions ########
+
+    # Update function to return modified object
+    uploadAccNumFile <- function(file_path, seq_obj) {
+        accnum_data <- read_file(file_path)
+        seq_obj@seqs <- accnum_data
+        return(seq_obj)
+    }
+
+    uploadFastaFile <- function(file_path, seq_obj) {
+        fasta_data <- read_file(file_path)
+        seq_obj@seqs <- fasta_data
+        return(seq_obj)
+    }
+
+    uploadMSAFile <- function(file_path, seq_obj) {
+        msa_data <- read_file(file_path)
+        seq_obj@seqs <- msa_data
+        return(seq_obj)
+    }
+
+    uploadBlastFile <- function(file_path, blast_obj) {
+        blast_obj@df <- file_path
+        return(blast_obj)
+    }
+
+    uploadIPRScanFile <- function(file_path, ipr_obj) {
+        ipr_obj@df <- file_path
+        return(ipr_obj)
+    }
+
+    # Now modify the calling section to reassign the modified objects
+    if (!is.null(file_paths$accnum)) {
+        sequence_upload_data <- uploadAccNumFile(file_paths$accnum,
+                                                 sequence_upload_data)
+    }
+    if (!is.null(file_paths$fasta)) {
+        sequence_upload_data <- uploadFastaFile(file_paths$fasta,
+                                                sequence_upload_data)
+    }
+    if (!is.null(file_paths$msa)) {
+        sequence_upload_data <- uploadMSAFile(file_paths$msa,
+                                              sequence_upload_data)
+    }
+    if (!is.null(file_paths$blast)) {
+        blast_upload_data <- uploadBlastFile(file_paths$blast,
+                                             blast_upload_data)
+    }
+    if (!is.null(file_paths$iprscan)) {
+        ipr_upload_data <- uploadIPRScanFile(file_paths$iprscan,
+                                             ipr_upload_data)
+    }
+
+    # Validation of inputs
+    fasta_data <- read_file(file_paths$fasta)
+    validate_and_process_inputs <- function(evalue, fasta_data) {
```

Comment:
rename func? Is this to validate all inputs or some subset, e.g., fasta inputs? Could rename accordingly.

Reply:
@awasyn, if you have addressed any comment, feel free to `Resolve conversation`. That way, you can keep track of a finite no. of unresolved comments alone.